lib/matplotlib/dates.py

```diff
@@ -1328,11 +1369,24 @@ def get_locator(self, dmin, dmax):
                 byranges[i] = None
                 continue
 
+            # Compute at runtime the size of date label with given format
+            if (axis_name == 'x'):
+                datelen_ratio = \
+                    len(dmin.strftime(self.eachtick[freq])) * font_ratio
+            else:
+                datelen_ratio = 1 * font_ratio
+
             # Find the first available interval that doesn't give too many
             # ticks
             for interval in self.intervald[freq]:
-                if num <= interval * (self.maxticks[freq] - 1):
-                    break
+
+                # Using an estmation of characters per inch, reduce
+                # intervals untill we get no overlaps
+                apply_spread = (not spacing or
+                                ((num/interval) * datelen_ratio) <= maxwid)
+                if (num <= interval * (self.maxticks[freq] - 1)):
+                    if (apply_spread):
+                        break
```

Comment:
??? I don't understand why you only break if apply_spread=True here.

Reply:
Same reasoning for adding the rcparams, when users have rotated labels we shouldn't apply spread, it applies too much spacing.

Reply:
But you need "interval" to be correct for `byranges`, below.

Reply:
for example test_dates.py::def test_auto_date_locator(), this uses dummy axis to specifically check values returned from autodatelocator. It does not have any axis properties though. 

https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/tests/test_dates.py#L320